### ISSUE TITLE
Race condition taking place in the `security.py` test

### ISSUE THREAD

**Issue by yanazaeva** [wazuh/wazuh#11957]
|Wazuh repo branch|Component|
|---|---|
| master | Framework |

## Description

This problem was initially witnessed in the issue opened to update the current [`test_worker.py`](https://github.com/wazuh/wazuh/issues/9999) unit tests. When this test was updated, the checks were failing, but the failures were not related to anything modified in the file.

Locally, the tests were passing without any problem, thus it was decided to use [ the following image ](https://github.com/wazuh/wazuh-jenkins/tree/master/dockers/util/test_unit) to do further testing. Using this image, this error was replicated in the [branch](https://github.com/wazuh/wazuh/tree/feature/9999-add-worker-test) created for the issue managed above, but also in the [master](https://github.com/wazuh/wazuh/tree/master) branch. 

This problem could be related to a modification done in [this issue](https://github.com/wazuh/wazuh/issues/11556), as previously the RBAC database was created by importing the module, and currently, it is created by calling the `create_rbac_db()` function.

The aim of this issue is to investigate the possible reason behind this failure and put a solution to it. 

### Examples of the error encountered
```
_______________________ ERROR at setup of test_load_spec _______________________
/usr/local/lib/python3.9/site-packages/sqlalchemy/ext/declarative/clsregistry.py:294: in __call__
    x = eval(self.arg, globals(), self._dict)
<string>:1: in <module>
    ???
E   NameError: name 'Roles' is not defined
```
```
E           sqlalchemy.exc.InvalidRequestError: When initializing mapper mapped class Roles->roles, expression 'User' failed to locate a name ("name 'User' is not defined"). If this is a class name, consider adding this relationship() to the <class 'wazuh.rbac.orm.Roles'> class after both dependent classes have been defined.
```

The full trace can be found here: [roles_error.zip](https://github.com/wazuh/wazuh/files/7939968/roles_error.zip)

### wazuh/wazuh
- Unit tests without failures. Updated and/or expanded if there are new functions/methods/outputs:
  - [x] Cluster (`framework/wazuh/core/cluster/tests/` & `framework/wazuh/core/cluster/dapi/tests/`)
  - [x] Core (`framework/wazuh/core/tests/`)
  - [x] SDK (`framework/wazuh/tests/`)
  - [x] RBAC (`framework/wazuh/rbac/tests/`)
  - [x] API (`api/api/tests/`)
- API tavern integration tests without failures. Updated and/or expanded if needed (`api/test/integration/`):
  - [x] Affected tests 
  - [x] Affected RBAC (black and white) tests
- [x] Review integration test mapping using the script (`api/test/integration/mapping/integration_test_api_endpoints.json`)
- [x] Review of spec.yaml examples and schemas (`api/api/spec/spec.yaml`)
- [x] Review exceptions remediation when any endpoint path changes or is removed (`framework/wazuh/core/exception.py`)
- [x] Changelog (`CHANGELOG.md`)

### wazuh/wazuh-documentation
- [x] Migration from 3.X for changed endpoints (`source/user-manual/api/equivalence.rst`)
- [x] Update RBAC reference with new/modified actions/resources/relationships (`source/user-manual/api/rbac/reference.rst`)

**Comment by vicferpoy** [comment]
# Issue update

After trying to reproduce this error under controlled environments without success, we have decided to refactor the affected tests.

It seems that the error only occurs in the virtualized environments we use for testing and not only the ORM code is related, since `4.3` has no errors and that part of the code is the same. Later changes on the framework tests for the `master` branch must have something to do with this.

**Comment by yanazaeva** [comment]
## Issue update

This issue was reopened as this problem is still happening at the `master` branch. I will leave here attached a file where this problem is seen. This was the only terminal used, no other changes rather than the ones captured here were implemented.

[terminal_output.zip](https://github.com/wazuh/wazuh/files/8068911/terminal_output.zip)

**Comment by Selutario** [comment]
This error happened again here:
- https://github.com/wazuh/wazuh/pull/17670/checks?check_run_id=14661049705

Full output: [B43842_pytest_framework.txt](https://github.com/wazuh/wazuh/files/11908465/B43842_pytest_framework.txt)

**Comment by nico-stefani** [comment]
# Issue update

The issue was reponed because we found a test failure in the same way but this time is occurring in [test_prepocessor.py](https://github.com/wazuh/wazuh/blob/master/framework/wazuh/rbac/tests/test_preprocessor.py).

The solution, proposed above, is working but is only limited to the tests that use the [db_setup](https://github.com/wazuh/wazuh/blob/master/framework/wazuh/tests/test_security.py#L71) fixture defined in `test_security.py`. 

There are several definitions of the `db_setup` fixture. So, I propose modifying them to implement the solution that fixes the problem to solve the most recent occurrence and prevent future ones in the other test files.

Fixture definitions:

* [test_proprocessor.py](https://github.com/wazuh/wazuh/blob/318f1aca5f764b7f4b5608798d83140bd317dfe9/framework/wazuh/rbac/tests/test_preprocessor.py#L19)
* [test_orm.py](https://github.com/wazuh/wazuh/blob/318f1aca5f764b7f4b5608798d83140bd317dfe9/framework/wazuh/rbac/tests/test_orm.py#L26)
* [test_default_configuration.py](https://github.com/wazuh/wazuh/blob/318f1aca5f764b7f4b5608798d83140bd317dfe9/framework/wazuh/rbac/tests/test_default_configuration.py#L19)
* [test_decorators.py](https://github.com/wazuh/wazuh/blob/318f1aca5f764b7f4b5608798d83140bd317dfe9/framework/wazuh/rbac/tests/test_decorators.py#L23)
* [test_auth_context.py](https://github.com/wazuh/wazuh/blob/318f1aca5f764b7f4b5608798d83140bd317dfe9/framework/wazuh/rbac/tests/test_auth_context.py#L19)

We need to discuss with the team the proposed solution and see if we continue with this approach.

**Comment by nico-stefani** [comment]
# Issue Update

I tried to reproduce the failure locally to check if the proposed solution worked.

Also, I tried over different branches but only the tests fail within `feature/16067-security-events-webhook` where are changes in `framework/wazuh/rbac/default/policies.yaml`, `framework/wazuh/rbac/default/relationships.yaml` and `framework/wazuh/tests/data/security/rbac_catalog.yml`.

**Comment by nico-stefani** [comment]
I was porting the solution to the `db_setup` in `test_preprocessor.py` and testing it against the changes of `feature/16067-security-events-webhook`.

With the changes implemented in 8ae7638, the pipeline execution was successful.

[B44180_framework_html_report.tar.gz](https://github.com/wazuh/wazuh/files/11985549/B44180_framework_html_report.tar.gz)

We need to decide if we port this fix to all `db_setup` fixtures.
